asttokens/asttokens.py

```diff
@@ -33,18 +32,17 @@
   from .util import AstNode, TokenInfo
 
 
-class ASTTextBase(six.with_metaclass(abc.ABCMeta, object)):
+class ASTTextBase(metaclass=abc.ABCMeta):
   def __init__(self, source_text, filename):
-    # type: (Any, str) -> None
-    # FIXME: Strictly, the type of source_text is one of the six string types, but hard to specify with mypy given
-    # https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
-
+    # type: (str, str) -> None
     self._filename = filename
 
     # Decode source after parsing to let Python 2 handle coding declarations.
     # (If the encoding was not utf-8 compatible, then even if it parses correctly,
     # we'll fail with a unicode error here.)
-    source_text = six.ensure_text(source_text)
+    # FIXME: This is the only remaining usage for six. Do we still need this check?
+    #  If so, then we should copy the function here and remove dependency on six
+    source_text =  six.ensure_text(source_text)
```

Comment:
review this part

Reply:
Yes, I'd say we should still decode bytes to preserve compatibility. So the type hint needs to be amended. Also the "Decode source after parsing to let Python 2..." doesn't even make sense and needs updating.